From 79588dbfc9ff0830d86ba6347b947ecd76e8a981 Mon Sep 17 00:00:00 2001
From: Niko Tyni <ntyni@debian.org>
Date: Sun, 29 Jun 2014 12:08:15 +0300
Subject: [PATCH 1/2] Fix t/directive/perlloadmodule2.t

The third test (for $url/subdir) was failing because the absence of
an argument ($url?...) caused a warning that $r->arg was undefined in
perlloadmodule2.pm, and the "use warnings FATAL => 'all';" line saw to
it that the server returned 500/Internal Server Error in this case.

Author: Steve Hay <steve.m.hay@googlemail.com>
Origin: upstream, http://svn.apache.org/viewvc?view=revision&revision=1565579
---
 t/response/TestDirective/perlloadmodule2.pm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/t/response/TestDirective/perlloadmodule2.pm b/t/response/TestDirective/perlloadmodule2.pm
index 7c24be3..bab91ce 100644
--- a/t/response/TestDirective/perlloadmodule2.pm
+++ b/t/response/TestDirective/perlloadmodule2.pm
@@ -82,7 +82,7 @@ sub handler : method {
 
     my $s = $r->server;
 
-    if ($r->args eq 'srv') {
+    if (defined $r->args and $r->args eq 'srv') {
         my $srv_cfg = $self->get_config($s);
         $r->print("srv: @{ $srv_cfg->{MyMergeTest}||[] }");
     }
-- 
2.0.1

